### PR TITLE
Add publish workflow

### LINKED ISSUE #51: Create a release workflow
The release workflow will need two parts:
- [ ] publish the federated extension to PyPI using a token
- [ ] publish the source extension to NPM

### PR DESCRIPTION
**Pull request by krassowski**
Fixes #51. I based it on jupyterlab-git workflow.

### CODE REVIEW

**Comment by krassowski** [comment]
It seems that I do not have the privileges to add tokens though...

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        python-version: '3.x'
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip build twine
+    - name: Publish the Python package
+      env:
+        TWINE_USERNAME: __token__
+        TWINE_PASSWORD: ${{ secrets.PYPI_API_TOKEN }}
+      run: |
+        python -m build
```
Comment (fcollonval) [inline .github/workflows/publish.yml:30]: I did not know about that package - thanks

Quick question, it seems that sdist and wheel are not activated by default (cf [documentation](https://pypa-build.readthedocs.io/en/stable/)). Should it be activated?

**Review by fcollonval — COMMENTED** [review]
@krassowski I upgrade your rights. You should be able to add secrets now.

I got also a question regarding the `build` package.

File: .github/workflows/publish.yml (hunk 1 above)
Comment (krassowski) [reply .github/workflows/publish.yml:30]: I think that it implicitly works thanks to the build_meta specified in the pyproject.toml (see https://setuptools.readthedocs.io/en/latest/build_meta.html#how-to-use-it); the documentation is a bit scarce on this indeed (see https://github.com/pypa/setuptools/issues/1698)

Comment (fcollonval) [reply .github/workflows/publish.yml:30]: Thanks for documentation pointers.

**Pull request merged**